src/main/fc/cli.c

```diff
@@ -2211,8 +2211,8 @@ static void cliDshotProg(char *cmdline)
                 int command = atoi(pch);
                 if (command >= 0 && command < DSHOT_MIN_THROTTLE) {
                     if (escNumber == ALL_ESCS) {
-                        for (unsigned i = 0; i < getMotorCount(); i++) {
-                            pwmWriteDshotCommand(i, command);
+                        for (uint8_t index = 0; index < getMotorCount(); index++) {
```

Comment:
Well the size of an integer is 32 bit on STM32. So if you use an `uint8_t`, the compiler will have to add range checks. In a simple case like the above, they can probably be optimised away, because there is no way there could be an overrun. But over time, simple cases often grow into complex cases, and at some point the checks can't be optimised away again. If you use unsigned for loops from the get go, you'll never get into this situation.